### PR TITLE
Added more stubbed data in Platform plugin

### PR DESCRIPTION
**Pull request by Naveen33-am**
## Description

Updated Platform plugin to support stubbed OS, Hardware, Screen, CPU, GPU information with limited data 

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Linux   |    ✅     |
| iOS     |    ❌     |
| Android |    ❌     |
| Windows |    ❌     |

## Checklist

* [x] I have tested this ona Ubuntu OS
* [ ] I added the documentation in `README.md`
* [ ] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp [hunk 1]
```diff
@@ -13,13 +13,36 @@ namespace rnsplugin {
...
-    platformThread_(!platformManagerName_.empty()? platformManagerName_.c_str(): "RNSPlatformManager-UnknownID"),      //not used for now
-    client_(client) {}
+  platformThread_(!platformManagerName_.empty()? platformManagerName_.c_str(): "RNSPlatformManager-UnknownID"),      //not used for now
+  client_(client) {}
 
 RNSPlatformManagerInterface::~RNSPlatformManagerInterface(){
   // platformThread_ will  flush the all theee pending tasks in thread before exiting.
 }
 
+RNSPlatformManagerInterface::PlatformDevice::DeviceProcessUintInfo::DeviceProcessUintInfo(string lname, string lcores, string ltechnology, string lfrequency)
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: looks like typo error "DeviceProcessUnitInfo". Is it ?

Comment (shefayaparwin) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: Just like to understand, what does that "l" in input variables names denotes

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp [hunk 2]
```diff
@@ -13,13 +13,36 @@ namespace rnsplugin {
...
+    :name(lname),
+  cores(lcores),
+  technology(ltechnology),
+  frequency(lfrequency){}
+
+RNSPlatformManagerInterface::PlatformDevice::DeviceOsInfo::DeviceOsInfo(string osProvider, string osName, string osVersion)
+    : provider(osProvider),
+  name(osName),
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: can you correct the alignment of initializer list

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp [hunk 3]
```diff
@@ -13,13 +13,36 @@ namespace rnsplugin {
...
+  version(osVersion){}
+
+RNSPlatformManagerInterface::PlatformDevice::DeviceHardwareInfo::DeviceHardwareInfo(string hwManufacturer, string hwModel, string hwType)
+    :manufacturer(hwManufacturer),
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp:35]: can you correct the alignment of initializer list

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp [hunk 4]
```diff
@@ -13,13 +13,36 @@ namespace rnsplugin {
...
+  model(hwModel),
+  type(hwType){}
+
+RNSPlatformManagerInterface::PlatformDevice:: DeviceScreenInfo::DeviceScreenInfo(string screenDepth, int screenWidth, int screenHeight, string screenRefresh)
+    :depth(screenDepth),
+  width(screenWidth),
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: can you correct the alignment of initializer list

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp [hunk 5]
```diff
@@ -13,13 +13,36 @@ namespace rnsplugin {
...
+    : provider(osProvider),
+  name(osName),
+  version(osVersion){}
+
+RNSPlatformManagerInterface::PlatformDevice::DeviceHardwareInfo::DeviceHardwareInfo(string hwManufacturer, string hwModel, string hwType)
+    :manufacturer(hwManufacturer),
+  model(hwModel),
+  type(hwType){}
+
+RNSPlatformManagerInterface::PlatformDevice:: DeviceScreenInfo::DeviceScreenInfo(string screenDepth, int screenWidth, int screenHeight, string screenRefresh)
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: suggest screenRefreshRate over screenRefresh. If you are changing change it associated places and assignments as well

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp [hunk 6]
```diff
@@ -30,6 +30,41 @@ bool RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::isTesting() {
   return true;
 }
 
+shared_ptr<DeviceCPUInfo> RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::getCPUInfo() {
+  if(cpuInfo_ == nullptr) {
+    cpuInfo_ = make_shared<DeviceCPUInfo>("Intel","4","5nm","5Ghz");
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp]: can you change Ghz-> GHz , to keep the common indication

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp [hunk 7]
```diff
@@ -30,6 +30,41 @@ bool RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::isTesting() {
...
+  }
+  return cpuInfo_;
+}
+
+shared_ptr<DeviceGPUInfo> RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::getGPUInfo() {
+  if(gpuInfo_ == nullptr) {
+    gpuInfo_ = make_shared<DeviceGPUInfo>("Nvidia","8","6nm","2.4Ghz");
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp]: same Here Hhz-->GHz

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.h [hunk 8]
```diff
@@ -14,6 +14,11 @@ namespace rnsplugin {
 
 using namespace std;
 
+using DeviceCPUInfo = RNSPlatformManagerInterface::PlatformDevice::DeviceProcessUintInfo;
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/plugins/stub/PlatformManager.h]: I guess it means "DeviceProcessUnitInfo". If you are changing, change it all over

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.h [hunk 9]
```diff
@@ -14,6 +14,11 @@ namespace rnsplugin {
...
+using DeviceGPUInfo = RNSPlatformManagerInterface::PlatformDevice::DeviceProcessUintInfo;
+using DeviceOsInfo = RNSPlatformManagerInterface::PlatformDevice::DeviceOsInfo;
```
Comment (shefayaparwin) [inline ReactSkia/pluginfactory/plugins/stub/PlatformManager.h:19]: Will it be more right to name as DeviceOSInfo ?? If you are changing, change it all over

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.h [hunk 10]
```diff
@@ -37,12 +37,55 @@ class RNSPlatformManagerInterface {
...
+      string depth;
+      int width;
+      int height;
+      string refresh;
+    };
+
+    using DeviceCPUInfo = DeviceProcessUintInfo;
+    using DeviceGPUInfo = DeviceProcessUintInfo;
+
     string interfaceIdiom {"tv"};
```
Comment (prasanna-jeyam) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.h:81]: Is this used anywhere .?

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp [hunk 11]
```diff
@@ -30,6 +30,41 @@ bool RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::isTesting() {
...
+shared_ptr<DeviceOsInfo> RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::getOsInfo() {
+  if(osInfo_ == nullptr) {
+    osInfo_ = make_shared<DeviceOsInfo>("UbuntuDev","GNU/Linux","20.2");
+  }
+  return osInfo_;
+}
+
+shared_ptr<DeviceHardwareInfo> RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::getHardwareInfo() {
+  if(hardwareInfo_ == nullptr) {
+    hardwareInfo_ = make_shared<DeviceHardwareInfo>("SAGEM","7250","ARM");
```
Comment (prasanna-jeyam) [inline ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp]: I remember the SAGEM model as 7252s

**Review by prasanna-jeyam — CHANGES_REQUESTED** [review]

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.h (hunk 10 above)
Comment (munezbn) [reply ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.h:81]: Yes in the modules JS code

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.h (hunk 9 above)
Comment (munezbn) [reply ReactSkia/pluginfactory/plugins/stub/PlatformManager.h:19]: Other APIs also using camel case so we are using same for Os

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp (hunk 5 above)
Comment (munezbn) [reply ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: @Naveen33-am  you can change it

**Review by munezbn — APPROVED** [review]
Kindly change Refresh to RefreshRate

File: ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp [hunk 12]
```diff
@@ -36,11 +36,11 @@ RNSPlatformManagerInterface::PlatformDevice::DeviceHardwareInfo::DeviceHardwareI
     model(hwModel),
     type(hwType){}
 
-RNSPlatformManagerInterface::PlatformDevice:: DeviceScreenInfo::DeviceScreenInfo(string screenDepth, int screenWidth, int screenHeight, string screenRefresh)
+RNSPlatformManagerInterface::PlatformDevice:: DeviceScreenInfo::DeviceScreenInfo(string screenDepth, int screenWidth, int screenHeight, string screenRefreshRate)
     :depth(screenDepth),
     width(screenWidth),
     height(screenHeight),
-    refresh(screenRefresh){}
+    refresh(screenRefreshRate){}
```
Comment (munezbn) [inline ReactSkia/pluginfactory/interfaces/RNSPlatformManagerInterface.cpp]: refresh  --> refreshRate

**Review by chatrapathi06 — APPROVED** [review]

**Review by shefayaparwin — APPROVED** [review]

File: ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp [hunk 13]
```diff
@@ -30,6 +30,41 @@ bool RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::isTesting() {
...
+shared_ptr<DeviceOsInfo> RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::getOsInfo() {
+  if(osInfo_ == nullptr) {
+    osInfo_ = make_shared<DeviceOsInfo>("UbuntuDev","GNU/Linux","20.2");
+  }
+  return osInfo_;
+}
+
+shared_ptr<DeviceHardwareInfo> RNSPlatformManagerInterfaceImpl::PlatformDeviceImpl::getHardwareInfo() {
+  if(hardwareInfo_ == nullptr) {
+    hardwareInfo_ = make_shared<DeviceHardwareInfo>("SAGEM","7252S","ARM");
```
Comment (pprabhu-ng) [inline ReactSkia/pluginfactory/plugins/stub/PlatformManager.cpp:56]: Why is sagem mentioned here...

File: ReactSkia/core_modules/RSkPlatform.cpp [hunk 14]
```diff
@@ -67,12 +67,13 @@ folly::dynamic RSkPlatformModule::getConstants() {
     return folly::dynamic::object();
   }
   shared_ptr<RNSPlatformManagerInterface::PlatformDevice>  device = platformManagerHandle_->currentDevice();
+  shared_ptr<RNSPlatformManagerInterface::PlatformDevice::DeviceOsInfo> osInfo = device->getOsInfo();
```
Comment (pprabhu-ng) [inline ReactSkia/core_modules/RSkPlatform.cpp:70]: Do we need to check osInfo for null here, since this is coming from platform plugin

**Review by prasanna-jeyam — APPROVED** [review]

**Review by chatrapathi06 — APPROVED** [review]

**Review by munezbn — APPROVED** [review]

**Pull request merged**